runtime/vm/jvminit.c

```diff
@@ -2387,6 +2397,10 @@ static UDATA checkArgsConsumed(J9PortLibrary* portLibrary, J9VMInitArgs* j9vm_ar
 			} else if (HAS_MAPPING( j9vm_args, i) && (MAPPING_FLAGS( j9vm_args, i ) & INVALID_OPTION)) {
 				j9nls_printf(PORTLIB, J9NLS_ERROR, J9NLS_VM_INVALID_CMD_LINE_OPT, optString);
 			} else {
+				/* If ignoreUnrecognizedXXOptions is set to JNI_TRUE, we should ignore any options that start with -XX: */
+				if (ignoreUnrecongizedXXOptions && (0 == strncmp(optString, VMOPT_XX, 4))) {
```

Comment:
The hard coded 4 should be `sizeof(VMOPT_XX) - 1`. The `- 1` removes the trailing null which is included in the sizeof.